pallets/clusters/src/lib.rs

```diff
@@ -165,38 +169,40 @@ pub mod pallet {
 		/// Missing permission to perform an operation
 		NoPermission,
 		/// Invalid inputs
-		InvalidInputs,
+		InvalidInput,
 		/// Technical error not categorized
 		SystematicFailure,
 		/// The owner is not correct.
 		OwnerNotCorrect,
 		/// The member already exists in the cluster
 		MemberExists,
+		/// Too many members
+		MembersLimitReached,
 		/// Member not found in the cluster
 		MemberNotFound,
 		/// Account proxy already associated with the cluster account
 		AccountProxyAlreadyExists,
 		/// Too many proxies associated with the cluster
 		TooManyProxies,
+		/// InsufficientBalance
+		InsufficientFunds,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Create a **Cluster** passing a name as input.
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
-		pub fn create_cluster(origin: OriginFor<T>, name: Vec<u8>) -> DispatchResult {
+		pub fn create_cluster(origin: OriginFor<T>, name: BoundedVec<u8, T::NameLimit>) -> DispatchResult {
 			let who = ensure_signed(origin.clone())?;
 
-			let bounded_name: BoundedVec<_, T::NameLimit> =
-				name.clone().try_into().expect("cluster name is too long");
-			ensure!(!bounded_name.len().is_zero(), Error::<T>::InvalidInputs);
+			ensure!(!name.len().is_zero(), Error::<T>::InvalidInput);
 
 			let extrinsic_index = <frame_system::Pallet<T>>::extrinsic_index()
 				.ok_or(Error::<T>::SystematicFailure)?;
 			let current_block_number = <frame_system::Pallet<T>>::block_number();
 
 			let cluster_id = blake2_128(
-				&[current_block_number.encode(), name.clone(), extrinsic_index.clone().encode(), who.clone().encode()].concat(),
+				&[current_block_number.encode(), name.to_vec().clone(), extrinsic_index.clone().encode(), who.clone().encode()].concat(),
```

Comment:
Oh, I did not realize to_vec was creating a copy. Thank you!
I changed it everywhere now to use `Vec::from(bounded_vec)`, which underneath does 
```rust
fn from(x: BoundedVec<T, S>) -> Vec<T> {
	x.0
}
```
 I could not do `bounded_vec.0` myself since it threw the error `0 is a private field`.

I used it as a reference everywhere I could. Much cleaner now.
Many thanks!

Reply:
Ah I see that this passes the inner directly ok, can you comment that, I miseed from code review